### ISSUE TITLE
Start renaming immediately after extract code action

### ISSUE THREAD

**Issue by strowk** [zed-industries/zed#21870]
### Check for existing issues

- [X] Completed

### Describe the feature

When I extract code to function I would like to rename that function right away.

1. I select a bit of code (in typescript file) and press Ctrl+.
2. From drop-down I select Extract to inner function.. or Extract to method...
3. Code is extracted
4. In code editor now the new line with call of method appeared and is selected

Instead of having that line selected, I would like for Zed to start renaming refactoring of that new method, i.e the same as when I drop the selection, move cursor to method call (on its name) and press F2.

I believe that user experience would be better like this since name that is suggested by default (newMethod) is ~~almost~~ never ok. Hence I would recommend to start renaming right away.

The focus of Zed on AI features can also be useful here, though it is not a part of this FR, however it could be also possible in later changes to replace newMethod with AI generated name, while still give user ability to override it since renaming would start immediately. The most convenient way that this could be done is to provide a suggestion in the box where new name is being typed. So the story would go this way: I run extract method action and after extraction is done I see renaming box and my cursor is focused on it, inside of it there is no text, but there is a suggestion, that I can apply by pressing Tab and then Enter, or else Tab and change, then Enter, or simply type the name I want instead. Such story could be enabled by the current FR, which is only limited to starting renaming right after extraction.

### Environment

Zed: v0.164.2 (Zed)
OS: Linux X11 ubuntu 24.04
Memory: 60.6 GiB
Architecture: x86_64
GPU: AMD Radeon Graphics (RADV GFX1103_R1) || radv || Mesa 24.0.9-0ubuntu0.2

### If applicable, add mockups / screenshots to help present your vision of the feature

Now:
![image](https://github.com/user-attachments/assets/ead3ff43-9d70-4c0c-824b-d01de7409088)
Instead:
![image](https://github.com/user-attachments/assets/e4979b23-cfb9-47a2-b2bf-c3cf52dad6e1)

**Comment by notpeter** [comment]
> user experience would be better if extract method code action would immediately follow by starting renaming that extracted method (or function). Name that is suggested by default (newMethod) is almost never ok. Hence I would recommend to start renaming right away.

Can you rephrase this as a set of steps to reproduce (e.g. take this example file, with this language server, trigger this action, this happened / I wish this happened).

**Comment by strowk** [comment]
@notpeter  done. 

I did not provide file, as I don't believe that is relevant what code it is. I did not select language server. 
I mentioned that the issue appeared to me in Typescript, but I believe the behavior here should be consistent across the whole of Zed, so I don't believe it is a good idea to make it work like only for Typescript language server.
